### PR TITLE
Support internlm-xcomposer2-4khd-7b awq

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: lmdeploy/vl/model/xcomposer2.py [hunk 1]
```diff
@@ -61,7 +61,8 @@ def build_model(self):
 
         # additional components.
         with add_sys_path(self.model_path):
-            if config.architectures[0] == 'InternLM2ForCausalLM':
+            if config.architectures[0] in ('InternLM2ForCausalLM',
+                                           'InternLMXComposer2ForCausalLM'):
                 # internlm-xcomposer2-4khd-7b
                 from ixc_utils import HD_transform
                 self.HD_transform = HD_transform
```
Comment (irexyc) [inline lmdeploy/vl/model/xcomposer2.py:68]: 这里为什么要改呢，这样改的话，不是4khd的话要报错了。

Comment (AllentDan) [reply lmdeploy/vl/model/xcomposer2.py]: 这里其实 4khd 和普通的 xcomposer build 出的模型，save 以后都是 InternLMXComposer2ForCausalLM。我改成 try except 看看文件夹里有无 ixc_utils.py 判断了。你那边有更好的方法的话可以分享下

**Comment by lvhan028** [comment]
Please fix UT

File: lmdeploy/lite/quantization/activation/observer.py [hunk 2]
```diff
@@ -101,7 +101,8 @@ def observe(self, x: torch.Tensor, save_input: bool = False) -> None:
         """
         if self.observed:
             return
-        assert len(x.shape) == 3
+        if len(x.shape) != 3:
```
Comment (lvhan028) [inline lmdeploy/lite/quantization/activation/observer.py:104]: why?

File: lmdeploy/vl/model/xcomposer2.py [hunk 3]
```diff
@@ -61,12 +61,12 @@ def build_model(self):
 
         # additional components.
         with add_sys_path(self.model_path):
-            if config.architectures[0] == 'InternLM2ForCausalLM':
+            try:
```
Comment (lvhan028) [inline lmdeploy/vl/model/xcomposer2.py:64]: why not handle it one by one?

**Review by irexyc — APPROVED** [review]

**Comment by lvhan028** [comment]
@zhulinJulia24 may add internlm-xcomposer2-4khd-7b model into TC

**Comment by lvhan028** [comment]
The result is in chaos when tp > 1

**Comment by AllentDan** [comment]
> The result is in chaos when tp > 1

internlm-xcomposer2-vl-7b-4bit also met the problem. I'm unsure if the mix-precision computation in turbomind is OK for tp>1.
https://github.com/InternLM/lmdeploy/blob/main/src/turbomind/models/llama/LlamaLinear.h#L31

**Comment by lvhan028** [comment]
tp=1 works fine, but tp > 1 failed. I don't think that's the mix-precision issue.
@irexyc @lzhangzz any idea?

**Comment by lvhan028** [comment]
> tp=1 works fine, but tp > 1 failed. I don't think that's the mix-precision issue. @irexyc @lzhangzz any idea?

I'll leave this problem to another PR. @lvhan028

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**